x-pack/legacy/plugins/lens/public/app_plugin/plugin.tsx

```diff
@@ -93,9 +104,54 @@ export class AppPlugin {
             http: core.http,
           })
         );
+        const updateUrlTime = (urlVars: Record<string, string>): void => {
+          const decoded: RisonObject = rison.decode(urlVars._g) as RisonObject;
+          if (!decoded) {
+            return;
+          }
+          // @ts-ignore
+          decoded.time = data.query.timefilter.timefilter.getTime();
+          urlVars._g = rison.encode((decoded as unknown) as RisonObject);
+        };
+        const redirectTo = (
+          routeProps: RouteComponentProps<{ id?: string }>,
+          addToDashboardMode: boolean,
+          id?: string
+        ) => {
+          if (!id) {
+            routeProps.history.push('/lens');
+          } else if (!addToDashboardMode) {
+            routeProps.history.push(`/lens/edit/${id}`);
+          } else if (addToDashboardMode && id) {
+            routeProps.history.push(`/lens/edit/${id}`);
+            const url = context.core.chrome.navLinks.get('kibana:dashboard');
+            if (!url) {
+              return;
+            }
+            const lastDashboardAbsoluteUrl = url.url;
+            const lensUrl = `${getKibanaBasePathFromDashboardUrl(
+              lastDashboardAbsoluteUrl
+            )}/lens/edit/${id}`;
+            if (lastDashboardAbsoluteUrl && lensUrl) {
+              const urlVars = getUrlVars(lastDashboardAbsoluteUrl);
+              updateUrlTime(urlVars);
+              window.history.pushState({}, '', lensUrl);
```

Comment:
I think we generally avoid using pushState directly, and instead use a core service to do this, but I'm not sure. Might be worth looking into, anyway.

Reply:
I don't think it exists in the new platform. I think the `pushState` is our only option for now.